xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-default/src/main/java/org/xwiki/notifications/internal/DefaultModelBridge.java

```diff
@@ -224,6 +207,47 @@ public void setStartDateForUser(DocumentReference userReference, Date startDate)
         return preferences;
     }
 
+    /**
+     * Search for the XObject corresponding to the given notification preference in the given document. If no object
+     * is found, returns null.
+     *
+     * @param xWikiDocument the document to search in
+     * @param notificationPreference the notification preference
+     * @return the corresponding NotificationPreferences XObject
+     * @throws NotificationException if the base object could not be found
+     * @since 9.7RC1
+     */
+    private BaseObject findNotificationPreference(XWikiDocument xWikiDocument,
+            NotificationPreference notificationPreference) throws NotificationException
+    {
+        List<BaseObject> objects = xWikiDocument.getXObjects(NOTIFICATION_PREFERENCE_CLASS);
+
+        if (objects != null) {
+            Iterator<BaseObject> it = objects.iterator();
+            while (it.hasNext()) {
+                // Ensure that we’re dealing with the correct event type
+                BaseObject object = it.next();
```

Comment:
(minor): this could easily be replaced by `for (BaseObject: object : objects) {`